src/pages/Commits.tsx

```diff
@@ -52,21 +52,22 @@ export const Commits = () => {
   const recentlySelectedProject = Cookies.get(CURRENT_PROJECT);
   // Push default project to URL if there isn't a project in
   // the URL already and an mci-project-cookie does not exist.
-  useQuery<GetSpruceConfigQuery, GetSpruceConfigQueryVariables>(
-    GET_SPRUCE_CONFIG,
-    {
-      skip: !!projectId || !!recentlySelectedProject,
-      onCompleted({ spruceConfig }) {
```

Comment:
onCompleted will only invoked if useQuery fetches data over the network. So it is not a reliable way of triggering the redirect.